### DISCUSSION TITLE
Integrating Suspense 1 component at a time?

### DISCUSSION THREAD

**Discussion by Lanithane** [urql-graphql/urql#3421] [Q&A] [answered]
Hello,

My team would like to start using Suspense to lazy load components. Is it possible to do this per query, rather than at the client level?

It would make the most sense for us to start with migrating 1 query at a time to Suspense, rather than needing to migrate our entire app at once, however it does not look like `useQuery` accepts a `suspense` option- only with `createClient`.

**Comment by JoviDeCroock** [comment, marked as answer]
UseQuery does in the OperationContext parameter (second one), you have to memoize this object. https://formidable.com/open-source/urql/docs/api/core/#operationcontext

**Reply by Lanithane** [reply]
I will give this a shot today. Thank you

**Reply by Lanithane** [reply]
It looks like useQuery only accepts 1 parameter? Do you have any examples? I can't find any.

This is invalid code:
```
  const ctx: OperationContext = {
    suspense: true,
    requestPolicy: 'cache-and-network',
    url: urqlURL,
  }
  const ctxMemoized = React.useMemo(() => ctx, [])

  const [{ data, fetching: isQueryLoading, error }] = useQuery(
    {
      query: isAdmin || userID === currentUserID ? profileQuery : userQuery,
      variables: { id: userID },
      pause: !userID,
    },
    ctxMemoized,
  )
```

**Reply by JoviDeCroock** [reply]
sorry `context` is a property on the first argument https://formidable.com/open-source/urql/docs/basics/react-preact/#context-options the docs _do_ contain these

**Reply by Lanithane** [reply]
I got that all plugged in, but Suspense isn't working. It seems like nothing is happening- My suspense Fallback component is in place but it still errors out that data is undefined

**Reply by JoviDeCroock** [reply]
Hey, sorry about the delay in my response, this should have gotten fixed with https://github.com/urql-graphql/urql/pull/3427

**Reply by Lanithane** [reply]
Ty!
